playground/frontend/lib/modules/editor/components/editor_textarea.dart

```diff
@@ -133,44 +133,78 @@ class _EditorTextAreaState extends State<EditorTextArea> {
     focusNode.requestFocus();
     if (_codeController!.text.isNotEmpty) {
       _codeController!.selection = TextSelection.fromPosition(
-        TextPosition(offset: _findOffset()),
+        TextPosition(
+          offset: _getOffset(),
+        ),
       );
     }
   }
 
-  _findOffset() {
+  int _getOffset() {
+    int contextLine = _getIndexOfContextLine();
+    String pattern = _getPattern(_getQntOfStringsOnScreen());
+
+    if (pattern == '' || pattern == '}') {
+      return _codeController!.text.lastIndexOf(pattern);
+    }
+
     return _codeController!.text.indexOf(
-      _skipStrings(kNumberOfStringsToSkip),
-      _getPositionAfterImportsAndLicenses(widget.sdk),
+      pattern,
+      contextLine,
     );
   }
 
-  String _skipStrings(int qntOfStrings) {
-    List<String> strings = _codeController!.text
-        .substring(_getPositionAfterImportsAndLicenses(widget.sdk))
-        .split('\n');
+  String _getPattern(int qntOfStrings) {
+    int contextLineIndex = _getIndexOfContextLine();
+    List<String> stringsAfterContextLine =
+        _codeController!.text.substring(contextLineIndex).split('\n');
+
     String result =
-        strings.length > qntOfStrings ? strings[qntOfStrings] : strings.last;
-    if (result == '') {
-      return _skipStrings(qntOfStrings - 1);
-    } else {
-      return result;
+        stringsAfterContextLine.length + kAdditionalLinesForScrolling >
+                qntOfStrings
+            ? _getResultSubstring(stringsAfterContextLine, qntOfStrings)
+            : stringsAfterContextLine.last;
+
+    return result;
+  }
+
+  int _getQntOfStringsOnScreen() {
+    RenderBox? rBox =
```

Comment:
Is the question mark necessary here? It seems like rBox.size.height cannot be null.

Reply:
Thank you for the remark, I'll remove it.